contents/testing/security-testing.md

```diff
@@ -0,0 +1,91 @@
+
+<box id="article-toc">
+
+* [Introduction to Security Testing](#introduction-to-security-testing)
+    * [What Is Security Testing?‎](#what-is-security-testing)
+    * [Why Have Security Testing?](#why-have-security-testing)
+    * [How to Get Started with Security Testing?](#how-to-get-started-with-security-testing)
+        * [Phase 1: Before Development Begins](#phase-1-before-development-begins)
+        * [Phase 2: During Definition and Design](#phase-2-during-definition-and-design)
+        * [Phase 3: During Development](#phase-3-during-development)‎
+        * [Phase 4: During Deployment‎](#phase-4-during-deployment)
+        * [Phase 5: Maintenance and Operations](#phase-5-maintenance-and-operations)
+    * [Additional Resources‎](#additional-resources)
+</box>
+
+## What Is Security Testing?
+
+Security Testing is **a type of Software Testing** that uncovers vulnerabilities, threats, risks in a software application and prevents malicious attacks from intruders. The purpose of Security Tests is to identify all possible loopholes and weaknesses of the software system which might result in a loss of information, revenue, repute at the hands of the employees or outsiders of the Organization.
+
+The goal of security testing is to:
+- Identify the threats in the system.
+- To measure the potential vulnerabilities of the system.
+- To help in detecting every possible security risks in the system.
+- To help developers in fixing the security problems through coding.
+
+:fas-info-circle: Knowledge of Software Development Life-Cycles (SDLCs) would be useful but not necessary.
+
+![Generic SDLC Model](security-testing/Sec-Test-Fig-1.png "Generic SDLC Model")
+
+## Why Have Security Testing?
+
+A comprehensive security testing framework deals with validation across all layers of an application. Through this, the organization can evaluate their application code for vulnerabilities and take remedial measures for the same. Recently, many of the software development organizations have been making use of secure software development life cycle methodologies to ensure identification and rectification of vulnerability areas early on in the application development process.
+
+__Security Testing helps to achieve__:
+- Confidentiality of Information.
+- Integrity of Information.
+- Authenticity of Identity.
+- Authorisation: The process of determining that a requester is allowed to receive a service or perform an operation (e.g. Access Control).
+- Availability: Assuring information and communications services will be ready for use when expected.
+- Non-Repudiation: In reference to digital security, non-repudiation means to ensure that a transferred message has been sent and received by the parties claiming to have sent and received the message.
+
+## How to Get Started with Security Testing?
+
+As a subset of software testing, security testing operates in tandem with your SDLC. It is a conscious decision that needs to be made for effective application.
```

Comment:
not sure `for effective application` is a good way to describe it, how about `for a more secure application`?